app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -51,26 +51,34 @@ class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
     attr_accessor :client
 
     def verify_credentials(_auth_type = nil, options = {})
-      begin
-        # As the connect will only give a handle
-        # we verify the credentials via an actual operation
-        connect(options).inventory.list_feeds
-      rescue URI::InvalidComponentError
-        raise MiqException::MiqHostError, "Host '#{hostname}' is invalid"
-      rescue ::Hawkular::ConnectionException
-        raise MiqException::MiqUnreachableError, "Unable to connect to #{hostname}:#{port}"
-      rescue ::Hawkular::Exception => he
-        raise MiqException::MiqInvalidCredentialsError, 'Invalid credentials' if he.status_code == 401
-        raise MiqException::MiqHostError, 'Hawkular not found on host' if he.status_code == 404
-        raise MiqException::MiqCommunicationsError, he.message
-      rescue => err
-        $log.error(err)
-        raise MiqException::Error, 'Unable to verify credentials'
-      end
+      self.class.validate_connection(connect(options), options[:hostname], options[:port])
 
       true
     end
 
+    def self.validate_connection(connection, hostname = nil, port = nil)
+      # As the connect will only give a handle
+      # we verify the credentials via an actual operation
+      connection_rescue_block(hostname, port) do
+        connection.inventory.list_feeds
```

Comment:
@josejulio I'm not sure, but with Inventory V4 is list_feeds available?  This is not a problem for this PR but it may be a problem for applying it to the hawkular-1275 branch.

Reply:
Yes, but i don't think there is other way to test it at this point, "status" is unprotected, so it won't be good way to test the connection.